### PR TITLE
Add backLink behaviour for checkReport

### PR DESCRIPTION
**Pull request by sulthan-ahmed**
Previously on the /image step, if you clicked on the link then there would not be a backLink.  This addresses this issue

### CODE REVIEW

File: apps/rotm/behaviours/back-link-check-report.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+'use strict';
+
+module.exports = superclass => class extends superclass {
+  getBackLink(req, res, callback) {
+    let backLink = super.getBackLink(req, res, callback);
+      if (!backLink) {
+        backLink = 'image';
```
Comment (JoeChapman) [inline apps/rotm/behaviours/back-link-check-report.js]: Question: Do the 'backlinks' get prefixed with a `/`?

File: apps/rotm/behaviours/back-link-check-report.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+'use strict';
+
+module.exports = superclass => class extends superclass {
+  getBackLink(req, res, callback) {
+    let backLink = super.getBackLink(req, res, callback);
```
Comment (JoeChapman) [inline apps/rotm/behaviours/back-link-check-report.js]: `super.getBackLink` is asynchronous, the implementation should not return a value, but do stuff and then calls back with an error or `null` after an undetermined period of time.

Does `super.getBackLink` need to be async? If so, then you need to change the implementation of `getBackLink` and amend the tests

File: apps/rotm/behaviours/back-link-check-report.js (hunk 1 above)
Comment (sulthan-ahmed) [reply apps/rotm/behaviours/back-link-check-report.js]: yeah it seems it is added by the getBackLink in hof somewhere

Comment (JoeChapman) [reply apps/rotm/behaviours/back-link-check-report.js]: the template?

File: apps/rotm/behaviours/back-link-check-report.js (hunk 2 above)
Comment (JoeChapman) [reply apps/rotm/behaviours/back-link-check-report.js]: Have you tried running the app with this new behaviour?

Comment (sulthan-ahmed) [reply apps/rotm/behaviours/back-link-check-report.js]: so is that the case for all async functionality? i.e. they shouldn't return a value?

I just want `getBackLink` to return a value so I am guessing it shouldn't be async

File: apps/rotm/acceptance/features/back-link-check-your-report.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
+'use strict';
+
+const defaults = require('../defaults');
+
+Feature('Given I am on /image');
+
+// bugfix
```
Comment (JoeChapman) [inline apps/rotm/acceptance/features/back-link-check-your-report.js]: erm, what's this comment for? Remove it or elaborate substantially

File: apps/rotm/behaviours/back-link-check-report.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
+'use strict';
+
+module.exports = superclass => class extends superclass {
+  getBackLink(req, res) {
+    let backLink = super.getBackLink(req, res);
+      if (!backLink) {
```
Comment (JoeChapman) [inline apps/rotm/behaviours/back-link-check-report.js]: this condition is indented too far to the right.

File: apps/rotm/index.js [hunk 5]
```diff
@@ -47,7 +48,7 @@ module.exports = {
     },
     '/check-your-report': {
       prereqs: ['/image'],
-      behaviours: [require('hof-behaviour-summary-page'), 'complete', caseworkerEmailer],
+      behaviours: [require('hof-behaviour-summary-page'), 'complete', caseworkerEmailer, backLinkCheckReport],
```
Comment (JoeChapman) [inline apps/rotm/index.js]: I think the behaviour should be called `checkReportBackLink`.

File: apps/rotm/behaviours/back-link-check-report.js [hunk 6]
```diff
@@ -0,0 +1,11 @@
+'use strict';
+
+module.exports = superclass => class extends superclass {
+  getBackLink(req, res) {
+    let backLink = super.getBackLink(req, res);
```
Comment (JoeChapman) [inline apps/rotm/behaviours/back-link-check-report.js]: Where is `super.getBackLink`? Is `suer.getBackLink` sync or async? Can you point me to it please?

File: apps/rotm/acceptance/features/back-link-check-your-report.js (hunk 3 above)
Comment (sulthan-ahmed) [reply apps/rotm/acceptance/features/back-link-check-your-report.js]: It was an approach that I picked up from Lenny in r2r.  See here, https://github.com/UKHomeOffice/right-to-rent-check/search?utf8=%E2%9C%93&q=bugfix&type=

If a new dev was to review this codebase for the first time, they may question why we have such a basic test.  Adding the comment would help to explain there was a bug previously

File: apps/rotm/behaviours/back-link-check-report.js (hunk 6 above)
Comment (sulthan-ahmed) [reply apps/rotm/behaviours/back-link-check-report.js]: looks like it's sync 
https://github.com/UKHomeOfficeForms/hof-form-controller/blob/master/lib/controller.js#L83

does that mean in this example the callback was not required
https://github.com/UKHomeOffice/firearms/blob/ced343a97023bac42aea7e45ff4b5f568154ad5c/apps/common/controllers/address-loop.js#L15

File: apps/rotm/index.js (hunk 5 above)
Comment (sulthan-ahmed) [reply apps/rotm/index.js]: do you mind if I ask why? I added the prefix backLink because that's the issue at hand here but I'm not fussed about it

**Pull request merged**